File: src/kbs/baekjoon/silver/Baekjoon2485.java
package kbs.baekjoon.silver;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Baekjoon2485 {
	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int N=Integer.parseInt(br.readLine());
		int[] n_arr=new int[N];
		int[] n_length=new int[N-1];
		int gcd = Integer.MAX_VALUE;
		for(int i=0;i<N;i++) {
			n_arr[i]=Integer.parseInt(br.readLine());
			if(i>0) {n_length[i-1]=n_arr[i]-n_arr[i-1];}
		}
		gcd=n_length[0];
		for(int i=1;i<N-1;i++) {
			for(int j=Math.min(gcd, n_length[i]);j>=1;j--) {
				if(gcd%j==0 && n_length[i]%j==0) {gcd=j; break;}
			}
		}
		int result=0;
		for(int i=0;i<N-1;i++) {
			result+=n_length[i]/gcd-1;
		}
		System.out.println(result);
	}
}

File: src/ljh/baekjoon/silver/baekjoon_2485.py
# baekjoon 알고리즘 2485
# 2023.09.17 이주현

import sys
from math import gcd

num_of_trees = int(sys.stdin.readline())

trees_dist = []

tree = int(sys.stdin.readline())

for i in range (num_of_trees -1 ):
    next_tree = int(sys.stdin.readline())
    trees_dist.append(next_tree - tree)
    tree = next_tree
    
    
g = trees_dist[0]
for j in range(1, len(trees_dist)):
    g = gcd(g, trees_dist[j])

result = 0
for t in trees_dist:
    result += t // g - 1

print(result)